.github/workflows/check.yml

```diff
@@ -4,6 +4,9 @@ on:
   push:
     branches: [main]
   pull_request:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
```

Comment:
`github.head_ref` is the same for all pushes to the same PR. `github.run_id ` is the fallback for non-PRs, and is unique for every run (i.e. will never cause cancellations)